guides/advanced_guides/faceted_search.md

```diff
@@ -0,0 +1,172 @@
+  {
+      "id": "495925",
+      "title": "Doraemon the Movie:Nobita's Treasure Island",
+      "director": "Fujiko Fujio",
+      "genre": "Science fiction",
+      "poster": "https://image.tmdb.org/t/p/w1280/cmJ71gdZxCqkMUvGwWgSg3MK7pC.jpg",
+      "overview": "The story is based on Robert Louis Stevenson's Treasure Island novel.",
+      "release_date": 1520035200
+  },
+  {
+      "id": "329996",
+      "title": "Dumbo",
+      "director": "Tim Burton",
+      "genre": "Science fiction",
+      "poster": "https://image.tmdb.org/t/p/w1280/279PwJAcelI4VuBtdzrZASqDPQr.jpg",
+      "overview": "A young elephant, whose oversized ears enable him to fly, helps...",
+      "release_date": 1553644800
+  },
+  {
+      "id": "458723",
+      "title": "Us",
+      "director": "Jordan Peele",
+      "genre": "Horror",
+      "poster": "https://image.tmdb.org/t/p/w1280/ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
+      "overview": "Husband and wife Gabe and Adelaide Wilson take their...",
+      "release_date": 1552521600
+  },
+  ...
+]
+```
+
+You would declare faceted attributes as follows:
+
+```bash
+$ curl \
+  -X POST 'http://localhost:7700/indexes/movies/settings' \
+  --data '{
+      "attributesForFaceting": [
+          "director",
+          "genre"
+      ]
+  }'
+```
+
+## Querying On Faceted Attributes
+
+When performing a search, you can specify:
+
+### 1. The facets for which to retrieve the matching count
+
+`facets=[<facetName>, <facetName>, ...]`
+
+This attribute can take two values:
+
+- `[<facetName>, <facetName>, ...]` (Optional, array of strings, defaults to `null`)
+
+  An array of strings that contains the facets for which to retrieve the matching count. The number of remaining candidates for each specified facet is returned. If a facet name doesn't exist, it will be ignored.
+
+- `["*"]`
+
+  The `*` character can also be used. In that case, a count for all facets is returned.
+
+If the `facets` parameter has been set, the returned results will contain two additional fields:
+
+- `facets`: The number of remaining candidates for each specified facet.
+
+- `exhaustiveFacetsCount`:
+  Returns `true` if the above count is **exhaustive**.
+  Otherwise, returns `false` if the above count is **approximative**.
+
+### Example
+
+Given a movie database, suppose that you want to know how many Batman movies are classified as `Action and Adventure`, how many of them are classified as `Animation` and what the number of Batman movies per director is. You would use the following CURL command:
+
+```bash
+$ curl --get 'http://localhost:7700/indexes/movies/search' \
+    --data-urlencode 'q=Batman' \
+    --data-urlencode 'facets=["genre, "director]'
+```
+
+### 2. The facets to filter on
+
+`facetFilters=["facetName:facetValue"]` or `facetFilters=[["facetName:facetValue"]]`
+
+This attribute can take two types of array:
+
+- `["facetName:facetValue"]` (Optional, array of strings)
+- `[["facetName:facetValue"]]` (Optional, array of arrays of strings)
```

Comment:
```suggestion
- `["facetName:facetValue"]` (Optional, array of strings)
- `[["facetName:facetValue"]]` (Optional, array of arrays of strings)
- `["facetName1:facetValue1", ["facetName2:facetValue2"]]` (Optional, array of strings or arrays of strings)
```

Reply:
I am not sur about this part of the document.
We repeat what we already said just above this list and these array forms are useless: Doing an OR with one element is the same as doing an AND and imply a new useless deeper array, this is the same for the new mixed array form I propose...